### PR TITLE
Exit prob test enhance

### LINKED ISSUE #258: exit_probability_randomizer - add more tests
include bundles
include large alpha and check deterministic behavior

### PR DESCRIPTION
**Pull request by tsvisabo**
Fixes #258

### CODE REVIEW

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 1]
```diff
@@ -437,6 +438,275 @@ void test_cw_topology_two_inequal_tips(void) {
...
+  flex_trit_t transaction_1_trits[FLEX_TRIT_SIZE_8019];
+  flex_trit_t transaction_2_trits[FLEX_TRIT_SIZE_8019];
+  flex_trit_t transaction_3_trits[FLEX_TRIT_SIZE_8019];
+  flex_trit_t transaction_4_trits[FLEX_TRIT_SIZE_8019];
+
+  flex_trit_t ep_trits[FLEX_TRIT_SIZE_8019];
+
+  flex_trits_from_trytes(transaction_1_trits, NUM_TRITS_SERIALIZED_TRANSACTION,
+                         TX_1_OF_4_VALUE_BUNDLE_TRYTES, FLEX_TRIT_SIZE_8019,
+                         FLEX_TRIT_SIZE_8019);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Last two params are in terms of trytes

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 2]
```diff
@@ -437,6 +438,275 @@ void test_cw_topology_two_inequal_tips(void) {
...
+  flex_trits_from_trytes(transaction_2_trits, NUM_TRITS_SERIALIZED_TRANSACTION,
+                         TX_2_OF_4_VALUE_BUNDLE_TRYTES, FLEX_TRIT_SIZE_8019,
+                         FLEX_TRIT_SIZE_8019);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Last two params are in terms of trytes

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 3]
```diff
@@ -437,6 +438,275 @@ void test_cw_topology_two_inequal_tips(void) {
...
+  flex_trits_from_trytes(transaction_3_trits, NUM_TRITS_SERIALIZED_TRANSACTION,
+                         TX_3_OF_4_VALUE_BUNDLE_TRYTES, FLEX_TRIT_SIZE_8019,
+                         FLEX_TRIT_SIZE_8019);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Last two params are in terms of trytes

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 4]
```diff
@@ -437,6 +438,275 @@ void test_cw_topology_two_inequal_tips(void) {
...
+  flex_trits_from_trytes(transaction_4_trits, NUM_TRITS_SERIALIZED_TRANSACTION,
+                         TX_4_OF_4_VALUE_BUNDLE_TRYTES, FLEX_TRIT_SIZE_8019,
+                         FLEX_TRIT_SIZE_8019);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Last two params are in terms of trytes

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 5]
```diff
@@ -437,6 +438,275 @@ void test_cw_topology_two_inequal_tips(void) {
...
+
+  flex_trits_from_trytes(ep_trits, NUM_TRITS_SERIALIZED_TRANSACTION,
+                         BUNDLE_OF_4_TRUNK_TRANSACTION, FLEX_TRIT_SIZE_8019,
+                         FLEX_TRIT_SIZE_8019);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Last two params are in terms of trytes

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 6]
```diff
@@ -437,6 +438,275 @@ void test_cw_topology_two_inequal_tips(void) {
...
+  flex_trit_t tx_1_of_4_trits[FLEX_TRIT_SIZE_8019];
+  flex_trit_t tx_2_of_4_trits[FLEX_TRIT_SIZE_8019];
+  flex_trit_t tx_3_of_4_trits[FLEX_TRIT_SIZE_8019];
+  flex_trit_t tx_4_of_4_trits[FLEX_TRIT_SIZE_8019];
+
+  flex_trit_t ep_trits[FLEX_TRIT_SIZE_8019];
+
+  flex_trits_from_trytes(tx_1_of_4_trits, NUM_TRITS_SERIALIZED_TRANSACTION,
+                         TX_1_OF_4_VALUE_BUNDLE_TRYTES, FLEX_TRIT_SIZE_8019,
+                         FLEX_TRIT_SIZE_8019);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Last two params are in terms of trytes

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 7]
```diff
@@ -437,6 +438,275 @@ void test_cw_topology_two_inequal_tips(void) {
...
+  flex_trits_from_trytes(tx_2_of_4_trits, NUM_TRITS_SERIALIZED_TRANSACTION,
+                         TX_2_OF_4_VALUE_BUNDLE_TRYTES, FLEX_TRIT_SIZE_8019,
+                         FLEX_TRIT_SIZE_8019);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Last two params are in terms of trytes

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 8]
```diff
@@ -437,6 +438,275 @@ void test_cw_topology_two_inequal_tips(void) {
...
+  flex_trits_from_trytes(tx_3_of_4_trits, NUM_TRITS_SERIALIZED_TRANSACTION,
+                         TX_3_OF_4_VALUE_BUNDLE_TRYTES, FLEX_TRIT_SIZE_8019,
+                         FLEX_TRIT_SIZE_8019);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Last two params are in terms of trytes

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 9]
```diff
@@ -437,6 +438,275 @@ void test_cw_topology_two_inequal_tips(void) {
...
+  flex_trits_from_trytes(tx_4_of_4_trits, NUM_TRITS_SERIALIZED_TRANSACTION,
+                         TX_4_OF_4_VALUE_BUNDLE_TRYTES, FLEX_TRIT_SIZE_8019,
+                         FLEX_TRIT_SIZE_8019);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Last two params are in terms of trytes

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 10]
```diff
@@ -437,6 +438,275 @@ void test_cw_topology_two_inequal_tips(void) {
...
+
+  flex_trits_from_trytes(tx_1_of_2_trits, NUM_TRITS_SERIALIZED_TRANSACTION,
+                         TX_1_OF_2, FLEX_TRIT_SIZE_8019, FLEX_TRIT_SIZE_8019);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Last two params are in terms of trytes

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 11]
```diff
@@ -437,6 +438,275 @@ void test_cw_topology_two_inequal_tips(void) {
...
+
+  flex_trits_from_trytes(tx_2_of_2_trits, NUM_TRITS_SERIALIZED_TRANSACTION,
+                         TX_2_OF_2, FLEX_TRIT_SIZE_8019, FLEX_TRIT_SIZE_8019);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Last two params are in terms of trytes

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 12]
```diff
@@ -437,6 +438,275 @@ void test_cw_topology_two_inequal_tips(void) {
...
+
+  flex_trits_from_trytes(ep_trits, NUM_TRITS_SERIALIZED_TRANSACTION,
+                         BUNDLE_OF_2_TRUNK_TX, FLEX_TRIT_SIZE_8019,
+                         FLEX_TRIT_SIZE_8019);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Last two params are in terms of trytes

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 13]
```diff
@@ -437,6 +438,275 @@ void test_cw_topology_two_inequal_tips(void) {
...
+  int selections = 10;
+  for (size_t i = 0; i < selections; ++i) {
+    TEST_ASSERT(iota_consensus_exit_probability_randomize(
+                    &ep_randomizer, &epv, &out, ep, &tip) == RC_OK);
+    if (memcmp(tip.trits, tx1_of_4->hash, FLEX_TRIT_SIZE_243) == 0) {
+      selected_tip_count++;
+    }
+  }
+
+  // TEST_ASSERT_EQUAL_INT(selected_tip_count,selections);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Shouldn't this be uncommented ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c (hunk 13 above)
Comment (tsvisabo) [reply consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: yes
it should

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**